packages/core/src/guards.ts

```diff
@@ -18,18 +27,41 @@ export function stateIn<TContext, TEvent extends EventObject>(
   };
 }
 
-export function stateNotIn<TContext, TEvent extends EventObject>(
-  stateValue: StateValue
-): GuardPredicate<TContext, TEvent> {
+export function not<TContext, TEvent extends EventObject>(
+  condition: Condition<TContext, TEvent>
+): BooleanGuardObject<TContext, TEvent> {
   return {
-    type: 'xstate.guard',
-    name: '!In',
-    predicate: (_, __, { state }) => {
-      if (isString(stateValue) && isStateId(stateValue)) {
-        return state.configuration.every((sn) => sn.id !== stateValue.slice(1));
-      }
+    type: DEFAULT_GUARD_TYPE,
+    op: 'not',
+    children: [toGuard(condition)]
+  };
+}
 
-      return !state.matches(stateValue);
-    }
+export function and<TContext, TEvent extends EventObject>(
+  ...conditions: Array<Condition<TContext, TEvent>>
```

Comment:
wondering if this just shouldn't be an array from the start? we already accept arrays in `choose` and i don't think we use variadic parameter lists at many places (if at any), given that we treat those as arrays anyway it makes sense to me to just require it to be an array and skip the conversion step on our side

Reply:
Or the other way around - perhaps `choose` should just use the arguments as the array? Still trying to decide; I'll compare to some prior art.

Reply:
the best reference that i know is https://ramdajs.com/docs/#cond , other than that i don't have super strong preference for one over another

Some things that come to my mind:
- variadic functions make `.length` unusable, this is a somewhat esoteric need so kinda whatever on that
- the transpiled code is often worse as spread often creates a for loop over `arguments`

Reply:
Using arrays instead 👍